app/src/main/kotlin/com/mapbox/vision/teaser/replayer/ReplayModeFragment.kt

```diff
@@ -0,0 +1,190 @@
+                replay_fragment_title.text = requireContext().resources.getQuantityString(R.plurals.selected_items, count, count)
+            }
+        }
+    }
+
+    private fun initEditButton() {
+        edit_sessions_list.setOnClickListener {
+            changeMultiSelection(true)
+        }
+    }
+
+    private fun initSwipeRefreshLayout() {
+        swipe_refresh_sessions.setOnRefreshListener {
+            sessionsAdapter?.updateSessionsList()
+            swipe_refresh_sessions.isRefreshing = false
+        }
+    }
+
+    private fun initDoneEditButton() {
+        done_edit.setOnClickListener {
+            changeMultiSelection(false)
+        }
+    }
+
+    private fun onActivateMultiSelection() {
+        changeMultiSelection(true)
+        setMultiSelectionTitle()
+    }
+
+
+    private fun initDeleteSessionsButton() {
+        delete_sessions.setOnClickListener {
+            sessionsAdapter?.let {
+                val sessionsPath = arguments?.getString(ARG_PARAM_SESSIONS_PATH)
+                if (it.isMultiSelection && sessionsPath != null) {
+                    try {
+                        for (session in it.selectedItems) {
+                            val sessionsFolder = File("$sessionsPath/$session")
+                            FileUtils.deleteFolder(sessionsFolder)
+                        }
+                    } catch (ex: RuntimeException) {
+                        Toast.makeText(requireContext(), ex.message, Toast.LENGTH_LONG).show()
+                    }
+                    it.updateSessionsList()
+                    changeMultiSelection(false)
+                }
+            }
+        }
+    }
+
+    private fun onSessionClick(fileName: String) {
+        if (sessionsAdapter?.isMultiSelection == true) {
+            setMultiSelectionTitle()
+        } else {
+            val listener = requireActivity()
+            if (listener is OnClickModeItemListener) {
+                listener.onClickSessionItem(fileName)
+                requireActivity().onBackPressed()
+            }
+        }
+    }
+
+    private fun onCameraClick() {
+        val listener = requireActivity()
+        if (listener is OnClickModeItemListener) {
+            listener.onClickCamera()
+            requireActivity().onBackPressed()
+        }
+    }
+
+    private fun setMultiSelectionTitle() {
+        sessionsAdapter?.apply {
+            val count = getSelectedCount()
+            replay_fragment_title.text = requireContext().resources.getQuantityString(R.plurals.selected_items, count, count)
+        }
+    }
+
+    private fun changeMultiSelection(activate: Boolean) {
+        if (activate) {
+            sessionsAdapter?.activateMultiSelection()
+            edit_sessions_list.visibility = GONE
+            select_all.visibility = VISIBLE
+            delete_sessions.visibility = VISIBLE
+            record_session.visibility = GONE
+            back_button.visibility = GONE
+            done_edit.visibility = VISIBLE
```

Comment:
Move to some function? And unify with code in `else` branch.

Reply:
you mean extract lines from if ... and else.. branches to separate functions?

Reply:
We have `view.show()/view.hide()` utils that's a shorthand for `.visibility = `

Reply:
I mean kindof `setActivateVisibilityState(visible: Boolean)` and put it all there.

Reply:
will change the visibility to show()\hide()\invisible()